### PR TITLE
Update sshj to 0.30.0 and improve algorithm order

### PR DESCRIPTION
**Pull request by fmeum**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [x] Dependency updates

## :scroll: Description

Updates sshj to 0.30.0, which brings support for rsa-sha2-* key types
and bugfixes related to RSA certificates and Android Keystore backed
keys.

Along the way, this improves the algorithm preferences to be consistent
with the Mozilla Intermediate SSH configuration (as far as possible,
given that most certificate types and some encryption algorithms are
not yet supported).

We also add "ext-info-c" to the kex algorithm proposal to work around
certain kinds of "user agent sniffing" that limits the support of
rsa-sha2-* key types.

## :green_heart: How did you test it?
Not yet, but will later. Please also check for regressions.

## :pencil: Checklist

- [x] I formatted the code with the IDE's reformat action (Ctrl + Shift + L/Cmd + Shift + L)
- [x] I reviewed submitted code
- [ ] I added a [CHANGELOG](CHANGELOG.md) entry if applicable

### CODE REVIEW

**Review by msfjarvis — COMMENTED** [review]
LGTM, will approve once I can test.

**Comment by fmeum** [comment]
> LGTM, will approve once I can test.

Thanks! ed25519, GitHub and old-style key files should be our main focus.

If it's okay with you, I would not do a formal merge for the two big PRs. Two much has changed in the meantime, I would rather force push a handcrafted updated version. But that of course has time since it will only make it into 1.12.0.

**Comment by msfjarvis** [comment]
> > LGTM, will approve once I can test.
> 
> Thanks! ed25519, GitHub and old-style key files should be our main focus.

Added a note in my to-do to check those.

> If it's okay with you, I would not do a formal merge for the two big PRs. Two much has changed in the meantime, I would rather force push a handcrafted updated version. But that of course has time since it will only make it into 1.12.0.

Yeah absolutely feel free to do whatever you find most comfortable.

**Comment by fmeum** [comment]
Tried different key types with GitHub, everything worked well, including rsa-sha2-* keys. @msfjarvis If you could test that 1) old style key files are handled correctly, even if their passphrase is mistyped first and 2) that manually imported ed25519 keys work and maybe 3) that ssh-rsa keys work with a few of your "broken" servers, I think we would be done here.

I will try to get the other two PRs ready to merge right after the August release.

**Comment by msfjarvis** [comment]
> Tried different key types with GitHub, everything worked well, including rsa-sha2-* keys. @msfjarvis If you could test that 1) old style key files are handled correctly, even if their passphrase is mistyped first and 2) that manually imported ed25519 keys work and maybe 3) that ssh-rsa keys work with a few of your "broken" servers, I think we would be done here.

Just tested all three, seems to work perfectly.

**Review by msfjarvis — APPROVED** [review]

**Pull request merged**
